### DISCUSSION TITLE
Questions about Nginx FastCGI timeout Settings

### DISCUSSION THREAD

**Discussion by zhkugh** [serversideup/docker-php#491] [Q&A]
### Steps To Reproduce

1. Send a long-running request.
2. Observe the 504 timeout error.

#### What did you expect?
I hope to receive suggestions on how to better adjust the Nginx and PHP-FPM configurations to handle long-running requests.
#### What happened instead?
504 Gateway Time-out

### Affected Docker Images

serversideup/php:8.1-fpm-nginx

### Anything else?

`    # Pass "*.php" files to PHP-FPM
    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass 127.0.0.1:9000;

        # Timeout Setting
        fastcgi_connect_timeout 180s;
        fastcgi_send_timeout 180s;
        fastcgi_read_timeout 180s;
    }`

**Comment by jaydrogers** [comment]
Have you looked into setting these?

- `PHP_FPM_PROCESS_CONTROL_TIMEOUT`
- `PHP_MAX_EXECUTION_TIME`
- `NGINX_FASTCGI_TIMEOUT`

https://serversideup.net/open-source/docker-php/docs/reference/environment-variable-specification

Here's a related issue when we added NGINX_FASTCGI_TIMEOUT too https://github.com/serversideup/docker-php/issues/424
